### ISSUE TITLE
DateEdit resets when starting to type with 0

### ISSUE THREAD

**Issue by ivancavlek** [Megabit/Blazorise#3729]
**Describe the bug**
A DateEdit component resets once you start typing with 0, irrelevant if it is a day, month or a year.

**To Reproduce**
You can reproduce it on the demo page - https://bootstrapdemo.blazorise.com/tests/forms. In _Date_ field type 11.11.1111. Then try to change (day or month or year) by starting to type with a 0, e.g. 01.01.1111. The date resets. Time component seems to be working properly.

**Expected behavior**
A date should be allowed to be typed with a leading zero.

**Comment by stsrki** [comment]
I don't think we have much control over this behavior. If I try to debug and do the `console.log` values that I get from the input this is what I get

Fill the input with 1s

```
OnChangeHandler: 0001-11-11
OnChangeHandler: 0011-11-11
OnChangeHandler: 0111-11-11
OnChangeHandler: 1111-11-11
```

Hitting zero, we get

```
OnChangeHandler: 
```

Meaning that it is handled by either the Blazor framework or by the browser before it even reaches Blazorise. I can try to use JavaScript to prevent clearing the value if it's partially filled. More investigation is needed.

**Comment by stsrki** [comment]
Bad news. Event with JS `keydown` event the value is cleared

```
element.addEventListener("keydown", event => {
    console.log("keydown" + event.target.value);
});
```

I'm open to any suggestions at this moment!

**Comment by stsrki** [comment]
I would wait until I investigate more.
